btf/traversal.go

```diff
@@ -40,8 +40,9 @@ func children(typ Type, yield func(child *Type) bool) bool {
 	// Explicitly type switch on the most common types to allow the inliner to
 	// do its work. This avoids allocating intermediate slices from walk() on
 	// the heap.
+	var tags []string
 	switch v := typ.(type) {
-	case *Void, *Int, *Enum, *Fwd, *Float:
+	case *Void, *Int, *Enum, *Fwd, *Float, *declTag:
 		// No children to traverse.
```

Comment:
I'd add this:
```suggestion
		// No children to traverse.
		// declTags is declared as a leaf type since it's parsed into .Tags fields of other types
		// during unmarshaling.
```